MDANSE_GUI/Src/MDANSE_GUI/InputWidgets/MoleculePreviewWidget.py

```diff
@@ -0,0 +1,141 @@
+        self.update_text_label()
+        self.show_formula()
+
+    def update_text_label(self):
+        """Show the molecule information as text."""
+        info_text = f"Molecule name: {self.mol_name}\n"
+        for key, value in self.mol_info["atom_number"].items():
+            info_text += f"Number of {key} atoms: {value}\n"
+        info_text += f"Number of such molecules in trajectory: {self.mol_info['no_of_molecules']}\n"
+        self.text_label.setText(info_text)
+
+    def prepare_rdkit_molecule(self) -> Union[chem.RWMol, None]:
+        """Create an rdkit molecule from the selected atoms."""
+        if len(self.mol_info["atom_indices"]) > self.max_atom_limit:
+            self.image_label.clear()
+            self.image_label.setText("Molecule is too large for preview")
+            return None
+        self.image_label.setText("")
+        large_molecule = self.atom_database.chemical_system.rdkit_mol
+        submolecule = chem.RWMol()
+        mapping = {}
+        self.bond_mapping = {}
+        for index in self.mol_info["atom_indices"]:
+            new_index = submolecule.AddAtom(large_molecule.GetAtomWithIdx(index))
+            mapping[index] = new_index
+        for bond in self.mol_info["bond_list"]:
+            new_pair = mapping[bond[0]], mapping[bond[1]]
+            bond_index = submolecule.AddBond(new_pair[0], new_pair[1])
+            self.bond_mapping[bond_index] = new_pair
+        allchem.Compute2DCoords(submolecule)
+        return submolecule
+
+    def show_formula(self):
+        """Draw the molecule."""
+        submolecule = self.prepare_rdkit_molecule()
+        draw_options = draw.MolDrawOptions()
+        draw_options.addAtomIndices = True
+        pil_image = draw.MolToImage(submolecule, size=(600, 600), options=draw_options)
+        qt_image = ImageQt(pil_image)
+        pixmap = QPixmap.fromImage(qt_image)
+        self.image_label.setPixmap(pixmap)
+
+    def filter_bonds(self, selected_atoms: list[int]) -> list[int]:
+        """Find and return only the bonds that appear in the molecule."""
+        return [
+            key
+            for key, pair in self.bond_mapping.items()
+            if pair[0] in selected_atoms and pair[1] in selected_atoms
+        ]
+
+    def show_formula_with_selection(self, selected_atoms: list[int]):
+        """Draw the formula and highlight the selected atoms and bonds."""
+        submolecule = self.prepare_rdkit_molecule()
+        optional_bonds = self.filter_bonds(selected_atoms)
+        driver = draw.rdMolDraw2D.MolDraw2DCairo(600, 600)
+        draw_options = draw.MolDrawOptions()
+        draw_options.addAtomIndices = True
+        driver.SetDrawOptions(draw_options)
+        draw.rdMolDraw2D.PrepareAndDrawMolecule(
+            driver,
+            submolecule,
+            highlightAtoms=selected_atoms,
+            highlightBonds=optional_bonds,
+        )
+        png_data = driver.GetDrawingText()
+        image = QImage.fromData(png_data)
```

Comment:
```suggestion
        image = QImage.fromData(png_data).copy()
```
This fixes the crash on Windows.